addons/website/static/src/js/theme_preview_form.js

```diff
@@ -0,0 +1,95 @@
+odoo.define('website.theme_preview_form', function (require) {
+"use strict";
+
+var FormController = require('web.FormController');
+var FormView = require('web.FormView');
+var viewRegistry = require('web.view_registry');
+var core = require('web.core');
+var _t = core._t;
+var qweb = core.qweb;
+
+var ThemePreviewController = FormController.extend({
+    events: Object.assign({}, FormController.prototype.events, {
+        'click .o_use_theme': '_onUseThemeClick',
+        'click .o_switch_theme': '_onSwitchThemeClick',
+        'click .o_switch_mode_button': '_onSwitchButtonClick',
+    }),
+
+    init: function () {
+        this._super.apply(this, arguments);
+        this.$switchButton = false;
+    },
+    // -------------------------------------------------------------------------
+    // Public
+    // -------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    renderButtons: function ($node) {
+        var $previewButton = $(qweb.render('ThemePreview.Buttons'));
+        $node.html($previewButton);
+        this.$switchButton = this._renderSwitchButton();
+        $node.find('.o_switch_mode_button').replaceWith(this.$switchButton);
+    },
+    /**
+     * @override
+     */
+    start: function () {
```

Comment:
Should be above public functions (the "lifecycle" methods are put in lifecycle order at the top)